runtime/vm/resolvesupport.cpp

```diff
@@ -1368,18 +1446,46 @@ resolveVirtualMethodRefInto(J9VMThread *vmStruct, J9ConstantPool *ramCP, UDATA c
 			cpShapeDescription = J9ROMCLASS_CPSHAPEDESCRIPTION(cpClass->romClass);
 			lookupOptions |= J9_LOOK_CLCONSTRAINTS;
 		}
-		
-#if defined(J9VM_OPT_METHOD_HANDLE)
-		/*
-		 * Check for MH.invoke and MH.invokeExact.
-		 *
-		 * Methodrefs corresponding to those methods already have their methodIndex set to index into
-		 * cpClass->methodTypes. We resolve them by calling into MethodType.fromMethodDescriptorString()
-		 * and storing the result into the cpClass->methodTypes table.
-		 */
+
+#if defined(J9VM_OPT_METHOD_HANDLE) || defined(J9VM_OPT_OPENJDK_METHODHANDLE)
 		if ((NULL != cpShapeDescription)
 		&& (resolvedClass == J9VMJAVALANGINVOKEMETHODHANDLE(vm))
 		) {
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+			/**
+			 * Check for MH intrinsic methods
+			 *
+			 * Modify the signature to avoid signature mismatch due to varargs
+			 * These methods have special INL send targets
+			 */
+			J9UTF8 *nameUTF = J9ROMNAMEANDSIGNATURE_NAME(nameAndSig);
+			U_8* initialMethodName = J9UTF8_DATA(nameUTF);
+			U_16 initialMethodNameLength = J9UTF8_LENGTH(nameUTF);
+
+			if (isMethodHandleINL(initialMethodName, initialMethodNameLength)) {
+				J9UTF8 *modifiedMethodName = (J9UTF8 *)(nameAndNAS + sizeof(J9ROMNameAndSignature));
+				J9UTF8 *modifiedMethodSig = (J9UTF8 *)(nameAndNAS + sizeof(nameAndNAS) - sizeof(J9UTF8));
+				memset(nameAndNAS, 0, sizeof(nameAndNAS));
+
+				/* Create new J9ROMNameAndSignature */
+				nameAndSig = (J9ROMNameAndSignature *)nameAndNAS;
+				NNSRP_SET(nameAndSig->name, modifiedMethodName);
+				NNSRP_SET(nameAndSig->signature, modifiedMethodSig);
+
+				modifiedMethodName->length = initialMethodNameLength;
+				memcpy(modifiedMethodName->data, initialMethodName, initialMethodNameLength);
+
+				/* Set flag for partial signature lookup. Signature length is already initialized to 0. */
```

Comment:
Similar comment applies here as well about using DIRECT option.

Also, names like "modifiedX" should in some way, modify X.  Otherwise the code is confusing